include/openssl/core_numbers.h

```diff
@@ -12,6 +12,7 @@
 
 # include <stdarg.h>
 # include <openssl/core.h>
+# include <openssl/err.h>
```

Comment:
Providers might not be linked against libcrypto Until now such providers only had to include a very minimal set of OpenSSL headers to get some structure definitions and macro definitions etc. I assume the reason we need to add `err.h` here is to pick up the definition of ERR_STRING_DATA - but I'm not sure it is a good idea. It starts to feel like a slippery slope.

Reply:
OK, so how would you solve the problem? (and yes, you assumed correctly)
An option would be to use OSSL_ITEM instead, would that be suitable? Considering provider_core.c makes a copy of the array anyway, it might actually not be so bad (although I consider the need to copy an abomination, but here we are...)

Reply:
OSSL_ITEM would make a lot of sense since its something that providers are already exposed to. It even has this comment next to the structure definition:
````
 * This is used whenever we need to pass things like a table of error reason
 * codes <-> reason string maps, parameter name <-> parameter type maps, ...
````

Reply:
Done